docs/overrides.rst

```diff
@@ -0,0 +1,167 @@
+.. _`overrides`:
+
+=========
+Overrides
+=========
+
+Using Page Objects is a great way to isolate the extraction logic from the crawling logic.
+As a side effect, it is now pretty easy to **create a generic spider with a common crawling logic
+that works across different sites**. The unique missing requirement is to be able to
+configure different Page Objects for each different site, because the extraction logic
+surely changes from site to site.
+This is exactly the functionality that overrides provides.
+
+Let's see is with an example. The following is a a spider that extract books
+from `books.toscrape.com <http://books.toscrape.com/>`_ using Page Objects.
+
+.. code-block:: python
+
+    import scrapy
+    from web_poet import ItemWebPage, WebPage
+    from scrapy_poet import callback_for
+
+
+    class BookListPage(WebPage):
+
+        def book_urls(self):
+            return self.css('.image_container a::attr(href)').getall()
+
+
+    class BookPage(ItemWebPage):
+
+        def to_item(self):
+            return {
+                'url': self.url,
+                'name': self.css("title::text").get(),
+            }
+
+
+    class BooksSpider(scrapy.Spider):
+        name = 'books_spider'
+        start_urls = ['http://books.toscrape.com/']
+
+        def parse(self, response, page: BookListPage):
+            for url in page.book_urls():
+                yield response.follow(url, callback_for(BookPage))
+
+Note that the crawling logic is pretty simple and straightforward:
+
+1. Extract all books urls from the listing page
+2. For each book URL found in the step 1, fetch the page and extract the resultant item
+
+This logic should work without any change for different books sites because
+having pages with lists of books and then detail pages for the individual book is
+such a common way of structuring sites.
+
+Let's reuse the very same ``BooksSpider`` spider but now to extract books from a different
+page: `bookpage.com/reviews <https://bookpage.com/reviews>`_.
+
+We cannot reuse the Books to Scrape Page Objects in this case, so we have
+to implement new ones:
+
+.. code-block:: python
+
+    class BPBookListPage(WebPage):
+
+        def book_urls(self):
+            return self.css('.article-info a::attr(href)').getall()
+
+
+    class BPBookPage(ItemWebPage):
+
+        def to_item(self):
+            return {
+                'url': self.url,
+                'name': self.css(".book-data h4::text").get().strip(),
+            }
+
+The last step is configuring ``scrapy-poet`` so that these Page Objects
+are used for Book Page URLs. This can be done by configuring
+``SCRAPY_POET_OVERRIDES`` in the Scrapy ``settings.py`` configuration file.
+
+.. code-block:: python
+
+    SCRAPY_POET_OVERRIDES = {
+        "bookpage.com": {
+            BookListPage: BPBookListPage,
```

Comment:
If the subclass above makes sense, this mapping is not required, just need to list the new POs

Reply:
Autodetecting from the inheritance has its caveats. Every single class in hierarchy up to `Injectable` is a candidate to be the overridden class. How to select it? Using just the parent of a type? I think this is not enough, because you might use inheritance because of other reasons not related with overrides (e.g. creating a base class that contains some functionality but that still is abstract and needs requires create subclasses with the final implementation). 

Multiple inheritance could also be another problem. Imagine you define the following base classes: `Menu`, `Breadcrubms`, `Footer`. You might implement a PO `FooSitePO` that inherits for all of them, as it is providing all of them in one go. And you might be interested in overriding `Breadcrumbs` using `FooSitePo` but without being overriding `Footer` and `Menu`. With the current implementation, wich is explicit, this is not a problem.

Anyway, I wouldn't say I'm 100% sure we shouldn't go the detection from the inheritance way. Let's see if @kmike has some opinion here.